### PR TITLE
Introduce JSR-305 compatible nullability, remove `Optional`

### LINKED ISSUE #123: Returning empty values
From https://github.com/cloudevents/sdk-java/issues/119 and https://github.com/cloudevents/sdk-java/pull/108#discussion_r411193730:

What technique should we use, around the sdk, to notify the user of an eventual empty return value?

1. `Optional` getters
2. Have an annotation called `@Nullable` which notifies the user of an eventual null return value
3. Just explicit in the Javadoc that a method can return null

### PR DESCRIPTION
**Pull request by bsideup**
Closes #123.

Signed-off-by: Sergei Egorov <segorov@pivotal.io>

### CODE REVIEW

File: api/src/main/java/io/cloudevents/lang/Nullable.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package io.cloudevents.lang;
```
Comment (olegz) [inline api/src/main/java/io/cloudevents/lang/Nullable.java:1]: Should we include license header?

Comment (slinkydeveloper) [reply api/src/main/java/io/cloudevents/lang/Nullable.java:1]: Since most license headers are broken, let's do it in a followup with formatting

**Comment by slinkydeveloper** [comment]
@bsideup can i merge it?

**Comment by bsideup** [comment]
@slinkydeveloper yes, please 👍 Thanks for the quick review!

**Pull request merged**